capsules/extra/src/ieee802154/framer.rs

```diff
@@ -99,7 +99,52 @@ use kernel::ErrorCode;
 #[derive(Eq, PartialEq, Debug)]
 pub struct Frame {
     buf: &'static mut [u8],
-    info: FrameInfo,
+    info: FrameInfoWrap,
+}
+
+/// This enum wraps the `FrameInfo` struct and allows each sending type
+/// (Parse or Raw) to only store the relevant information. In the
+/// case of a Raw send, the `FrameInfo` struct is irrelevant as the
+/// packet has been fully formed by the userprocess. For a Raw send,
+/// we only require knowledge on the frame length. In the case of a
+/// Parse send, the wrapper provides all required frame header information.
+#[derive(Eq, PartialEq, Debug)]
+enum FrameInfoWrap {
+    Raw(usize),
+    Parse(FrameInfo),
+}
+
+impl FrameInfoWrap {
+    /// Obtain secured_length of the Frame
+    pub fn secured_length(&self) -> usize {
+        match self {
+            FrameInfoWrap::Parse(info) => info.secured_length(),
+            FrameInfoWrap::Raw(len) => *len,
+        }
+    }
+
+    /// Obtain unsecured_length of the Frame
+    pub fn unsecured_length(&self) -> usize {
+        match self {
+            FrameInfoWrap::Parse(info) => info.unsecured_length(),
+            FrameInfoWrap::Raw(len) => *len,
+        }
+    }
+
+    /// Fetcher of the FrameInfo struct for Parse sending. Panics if
+    /// called for Raw sending.
+    pub fn expect(&self) -> FrameInfo {
+        match self {
+            FrameInfoWrap::Raw(_) => {
+                // This should never be called for a Raw send. The Framer should never
+                // require information other than the Frame length for a Raw send. This
+                // warrants a panic condition as fetching the `FrameInfo` struct for a
+                // Raw send is undefined behavior.
+                panic!("FrameInfoWrap::Raw called when expecting FrameInfoWrap::Parse")
+            }
+            FrameInfoWrap::Parse(info) => *info,
+        }
+    }
```

Comment:
Similarly dubious about reusing the `expect` name. A little more so here because the API differs as well.

Reply:
I agree that this is misleading. I will update this naming to be `get_info`